api/controllers/checklist-controler.js

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable node/no-unsupported-features/es-syntax */
+const mongoose = require('mongoose');
+const CheckList = require('../models/checklist-model');
+const User = require('../models/user-model');
+
+const create = async (req, res) => {
+    try {
+        const { item_data } = req.body;
+        
+        if (!item_data) {
+            return res.status(422).json({message: 'Please, fill up all fields'});
+        }
+        
+        const newList = new CheckList({
+            author: mongoose.Types.ObjectId(),
+            creation_date: new Date(),
+            item_data
+        })
+        const list = await newList.save()
+        
+        res.status(201).json(list);
+    } catch (error) {
+        console.log("action-error", error);
+        res.status(500).json(error);
+    }
+};
+
+const getAll = async (req, res) => {
+    console.log("action-getAll");
+    try {
+        const checkLists = await CheckList.find();
+        const result = checkLists.map(doc => {
+            return {
+                id: doc.id,
+                author: doc.author.username,
```

Comment:
doc.author.username is undefined in your case 
just leave  **author: doc.author**